atomicfu/src/androidNative64BitMain/kotlin/kotlinx/atomicfu/parking/PosixParkingDelegator.kt

```diff
@@ -0,0 +1,72 @@
+package kotlinx.atomicfu.parking
+
+import kotlinx.atomicfu.AtomicBoolean
+import kotlinx.atomicfu.atomic
+import kotlinx.cinterop.*
+import kotlinx.cinterop.alloc
+import kotlinx.cinterop.free
+import kotlinx.cinterop.pointed
+import kotlinx.cinterop.ptr
+import platform.posix.*
+
+@OptIn(ExperimentalForeignApi::class)
+internal actual object PosixParkingDelegator : ParkingDelegator {
+    actual override fun createRef(): Any {
+        val combo = ParkingData(nativeHeap.alloc<pthread_mutex_t>().ptr, nativeHeap.alloc<pthread_cond_t>().ptr)
+        pthread_mutex_init(combo.mut, null)
+        pthread_cond_init(combo.cond, null)
+        return combo
+    }
+
+    actual override fun wait(ref: Any) {
+        if (ref !is ParkingData) throw IllegalArgumentException("ParkingDelegator got incompatible parking object")
+        pthread_mutex_lock(ref.mut)
+        while (!ref.wake.value) {
+            pthread_cond_wait(ref.cond, ref.mut)
+        }
+        pthread_mutex_unlock(ref.mut)
+    }
+
+    actual override fun timedWait(ref: Any, nanos: Long) {
+        if (ref !is ParkingData) throw IllegalArgumentException("ParkingDelegator got incompatible parking object")
+        val ts = nativeHeap.alloc<timespec>().ptr
```

Comment:
Minor: there's the https://kotlinlang.org/api/core/kotlin-stdlib/kotlinx.cinterop/mem-scoped.html function, which allows creating allocations more structurally.

```kotlin
memScoped {
  val ts = alloc<timespec>()
  clock_gettime(CLOCK_REALTIME.convert(), ts.ptr)
  // ...
  pthread_mutex_unlock(ref.mut)
} // `ts` gets deallocated when this lambda exits
```

Reply:
Done